model/EntityChecksumCacheService.php

```diff
@@ -0,0 +1,104 @@
+<?php
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2020 (original work) Open Assessment Technologies SA ;
+ */
+
+declare(strict_types=1);
+
+namespace oat\taoSync\model;
+
+use common_Exception;
+use common_persistence_KeyValuePersistence;
+use common_persistence_Manager;
+use InvalidArgumentException;
+use oat\generis\model\data\event\ResourceDeleted;
+use oat\oatbox\service\ConfigurableService;
+
+class EntityChecksumCacheService extends ConfigurableService
+{
+    public const SERVICE_ID = 'taoSync/EntityChecksumCacheService';
+    public const OPTION_PERSISTENCE = 'persistence';
+
+    private const PREFIX = 'entity-checksum-';
+
+    /**
+     * @param string $id
+     * @return bool|int|string|null
+     */
+    public function get(string $id)
```

Comment:
If we accept only string values in `set` method, do you think we coul set return type `?string`: return `null` if key does not exist and string value if exists?

Reply:
Updated